doc/getting_help.rst

```diff
@@ -1,43 +1,59 @@
 Getting help
 ============
 
+The NEST documentation has grown organically over the years and can be confusing to a newcomer. The following official documentation exists:
+
+* The `Sphinx documentation on Read the Docs <https://nest-simulator.readthedocs.io/en/stable/>`_. Several versions can be selected; make sure to use the one appropriate for your NEST version (which can be determined with ``nest.version()`` in Python). The "latest" version on Read the Docs mirrors the latest version on GitHub. This documentation is not built and installed with NEST by default - to do that, follow the instructions `here <https://github.com/nest/nest-simulator/tree/master/doc/README.rst>`_. There are **orphan pages** - i.e. pages that are not referenced from anywhere else within the documentation - which could be uncovered with Google. (TODO is this true, e.g. for SLI tutorials?) Besides providing a reference, it contains various tutorials and guides for aspects of NEST. This is the most relevant source of information for a new user who wants to use NEST with Python.
+* A legacy user documentation built by a homegrown tool called ``help_generator``. It is accessible by calling ``nest.helpindex()``/``nest.help("term")`` in Python or (for the latest stable version of NEST) `online <https://www.nest-simulator.org/helpindex/>`_. (TODO while this page is still up, clicking on something now yields a redirect to RTD, which doesn't seem ideal to me). This documentation doesn't describe PyNEST or NEST models (TODO is the latter a bug or feature?), but *SLI* (TODO anything else?), a programming language used to interact with the NEST kernel. Today, most end-users use PyNEST instead of programming in SLI. There are **naming collisions** between SLI commands and PyNEST functions which might be confusing. For example, ``nest.help("GetStatus")`` does **not** return the documentation of the ``GetStatus`` function in the PyNEST API, but of an equivalent command in SLI.
+* The Doxygen developer documentation, which documents the program code of NEST for developers. It can be built according to `these <https://nest-simulator.readthedocs.io/en/latest/documentation_workflow/developer_documentation_workflow.html>`_ instructions. As the reference files of the ``help_generator`` documentation are also compiled from Doxygen comments, it contains most of the ``help_generator`` content as well. Of course, you should develop for NEST based on the latest development version.
```

Comment:
```suggestion
The NEST documentation has grown tremendously over the years. The following material exists:

* `Read the Docs <https://nest-simulator.readthedocs.io/en/stable/>`_: This is the user 
  documentation of NEST. Besides providing a reference, it contains various NEST tutorials 
  and guides. Please make sure you select the documentation version appropriate for your 
  NEST version (as determined by ``nest.version()`` in Python). The "latest" version on 
  Read the Docs mirrors the latest version on GitHub.
* `Doxygen <https://www.doxygen.nl/index.html>`_: This is the developer documentation of 
  NEST. It can be built according to 
  :doc:`these instructions <documentation_workflow/developer_documentation_workflow>`. As 
  the reference files of the help generator documentation are also compiled from Doxygen 
  comments, it contains most of the help generator content as well. Of course, you should 
  develop for NEST based on the latest development version.
* `Help generator <https://www.nest-simulator.org/helpindex/>`_: This homegrown tool buils 
  legacy user user documentation that you can access by calling ``nest.helpindex()`` or 
  ``nest.help("term")`` in Python. For the latest stable version of NEST, you can also find 
  it `online <https://www.nest-simulator.org/helpindex/>`_. This documentation describes 
  SLI, a programming language used to interact with the NEST kernel.
* `Developer Space <https://nest.github.io/nest-simulator/>`_: This is a developer manual, 
  explaining the development infrastructure and workflow of NEST.
```